models/oneinch/ethereum/unoswap/oneinch_unoswap_v5_ethereum_trades.sql

```diff
@@ -0,0 +1,248 @@
+        incremental_strategy='merge',
+        unique_key = ['block_date', 'blockchain', 'project', 'version', 'tx_hash', 'evt_index', 'trace_address']
+    )
+}}
+
+{% set project_start_date = '2022-11-07' %} --for testing, use small subset of data
+
+WITH unoswap AS
+(
+    SELECT
+        call_block_number
+        ,output_returnAmount
+        ,amount
+        ,srcToken
+        ,pools
+        ,call_tx_hash
+        ,call_trace_address
+        ,call_block_time
+        ,contract_address
+    FROM
+        {{ source('oneinch_ethereum', 'AggregationRouterV5_call_unoswap') }}
+    WHERE
+        call_success
+        {% if is_incremental() %}
+        AND call_block_time >= date_trunc("day", now() - interval '1 week')
+        {% else %}
+        AND call_block_time >= '{{project_start_date}}'
+        {% endif %}
+
+    UNION ALL
+
+    SELECT
+        call_block_number
+        ,output_returnAmount
+        ,amount
+        ,srcToken
+        ,pools
+        ,call_tx_hash
+        ,call_trace_address
+        ,call_block_time
+        ,contract_address
+    FROM
+        {{ source('oneinch_ethereum', 'AggregationRouterV5_call_unoswapToWithPermit') }}
+    WHERE
+        call_success
+        {% if is_incremental() %}
+        AND call_block_time >= date_trunc("day", now() - interval '1 week')
+        {% else %}
+        AND call_block_time >= '{{project_start_date}}'
+        {% endif %}
+)
+, join_to_traces as
+(
+    SELECT
+        call_block_number
+        , output_returnAmount
+        , amount
+        , srcToken
+        , CASE
+            WHEN CAST(pools[array_size(pools) - 1] / POWER(2, 252) as int) & 2 != 0 --cast((pools[array_size(pools) - 1] / (POWER(2, 252))) as int) & 2 != 0
+            THEN '0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee'
+            ELSE to
+        END as dstToken
+        , pools
+        , call_tx_hash
+        , call_trace_address
+        , call_block_time
+        , contract_address
+        , from as taker
+    FROM
+    (
+        SELECT
+            unoswap.call_block_number
+            , unoswap.output_returnAmount
+            , unoswap.amount
+            , unoswap.srcToken
+            , unoswap.pools
+            , unoswap.call_tx_hash
+            , unoswap.call_trace_address
+            , unoswap.call_block_time
+            , unoswap.contract_address
+            , traces.to
+            , traces.from
+            , ROW_NUMBER() OVER (
+                PARTITION BY unoswap.call_tx_hash, unoswap.call_trace_address
+                ORDER BY traces.trace_address desc
+                ) as first_transfer_trace
+        FROM
+            unoswap
+        LEFT JOIN
+            {{ source('ethereum', 'traces') }} AS traces
+            ON traces.tx_hash = unoswap.call_tx_hash
+            AND traces.block_number = unoswap.call_block_number
+            AND traces.from != unoswap.contract_address
+            /*
+            --replace this logic with below COALESCE condition
```

Comment:
is this comment meant to be left in the source code?

Reply:
yeah, at least short term, as i wanted to indicate to 1inch team or other maintainers that i've changed the logic from v1 engine (which is what is commented out) to the simplified approach. it ideally will help any potential debugging. i can clean up the comment to indicate that